app/src/main/java/com/asfoundation/wallet/billing/partners/OemIdExtractorService.kt

```diff
@@ -1,20 +1,43 @@
 package com.asfoundation.wallet.billing.partners
 
 import android.content.Context
+import com.aptoide.injectorextractor.Extractor
 import io.reactivex.Single
+import java.io.File
 import java.util.*
 import java.util.zip.ZipFile
 
 
-class OemIdExtractorService(private val context: Context) {
+class OemIdExtractorService(context: Context) {
+
+  val extractorV1: OemIdExtractorV1 = OemIdExtractorV1(context)
+  val extractorV2: OemIdExtractorV2 = OemIdExtractorV2(context)
 
   fun extractOemId(packageName: String): Single<String> {
+    return extractorV2.extract(packageName)
+        .onErrorResumeNext(extractorV1.extract(packageName))
+  }
+}
+
+class OemIdExtractorV2(private val context: Context,
+                       private val extractor: Extractor = Extractor()) : IExtractOemId {
+  override fun extract(packageName: String): Single<String> {
+    return Single.fromCallable {
+      getPackageName(context, packageName)
+    }
+        .flatMap { sourceDir -> Single.fromCallable { extractor.extract(File(sourceDir)) } }
+        .map { t ->
+          t.split(",")[0]
```

Comment:
Use "it" instead.
In the previous one despite me saying that we should use "it", I think in this case it's better to keep the lambda to understand what the variable represents